grpc_api.go

```diff
@@ -0,0 +1,159 @@
+/*
+ * File: grpc_api.go
+ *
+ * Description: various funcs needed for the gRPC API
+ */
+
+package main
+
+import (
+    "strconv"
+    pb "./lclusterpb"
+    "golang.org/x/net/context"
+)
+
+//! API function to start a job
+/*
+ * @param     Context            given host context
+ * @param     StartJobRequest    object to hold the start job request
+ *
+ * @return    StartJobResponse   the server's response to the remote call
+ * @return    error              error message, if any
+ */
+func (s *LclusterdServer) StartJob(ctx context.Context,
+  r *pb.StartJobRequest) (*pb.StartJobResponse, error) {
+
+    // Variable declaration
+    response := &pb.StartJobResponse{}
+
+    // Cast the job request into a Job, then attempt to add it to the
+    // queue.
+    pid, err := etcdServer.addToGlobalQueue((*Job)(r))
+
+    // if any errors occur...
+    if err != nil || pid < 0 {
+
+        // set the pid to -1 and grab the error as a string
+        response.Pid = -1
+        response.Error = err.Error()
+
+        // then return the failed job creation via response
+        return response, err
+    }
+
+    // otherwise use the given pid and pass it back
+    response.Pid = pid
+    return response, nil
+}
+
+//! API function to check a job
+/*
+ * @param     Context            given host context
+ * @param     CheckJobRequest    object to hold the check job request
+ *
+ * @return    CheckJobResponse   the server's response to the remote call
+ * @return    error              error message, if any
+ */
+func (s *LclusterdServer) CheckJob(ctx context.Context,
+  cjr *pb.CheckJobRequest) (*pb.CheckJobResponse, error) {
+
+      // input validation
+      if cjr == nil || cjr.Pid < 1 {
+          return &pb.CheckJobResponse{Rc: -1}, errorf("CheckJob() --> " +
+                 "invalid input")
+      }
+
+      // grab the list of queued jobs
+      response, err := etcdServer.internal.Get(ctx, queue_dir)
+
+      // if an error occurs here, pass back a return code of 0, since for
+      // whatever reason, the server is unable to query jobs at this time
+      if err != nil {
+          return &pb.CheckJobResponse{Rc: 0}, err
+      }
+
+      // cycle thru all of the currently queued jobs
+      for _, j := range response.Kvs {
```

Comment:
ah yes, j is too much like a counter; thanks for pointing that out :)

i have since adjusted it to job

https://github.com/rbisewski/lclusterd/pull/2/commits/123b07f760048be3811a39e6d13f78182fa25564